SYCL/Basic/diagnostics/non-uniform-wk-gp-test.cpp

```diff
@@ -29,23 +30,23 @@ int test() {
                       << cl::sycl::endl;
           });
     });
-
+    std::cout << "Test failed: no exception thrown." << std::endl;
   } catch (sycl::runtime_error &E) {
     if (std::string(E.what()).find(
             "Non-uniform work-groups are not supported by the target device") !=
         std::string::npos) {
       std::cout << E.what() << std::endl;
       std::cout << "Test passed: caught the expected error." << std::endl;
-      return 0;
+      res = 0;
     } else {
       std::cout << E.what() << std::endl;
       std::cout << "Test failed: received error is incorrect." << std::endl;
-      return 1;
     }
   }
+  q.wait();
```

Comment:
Isn't it a bug that without queue::wait() the stream components are leaked on queue destruction?